scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,188 @@
+
+
+def is_published_uuid(uuid: str) -> str:
+    """
+    Check if the UUID provided corresponds to a published Azul project
+    :param uuid:    UUID of the project which contains the files
+    :return uuid:
+    """
+    r = rq.get(f"https://service.azul.data.humancellatlas.org/index/projects/{uuid}")
+    try:
+        r.raise_for_status()
+    except HTTPError:
+        raise NotPublishedUuid(r, uuid)
+    return uuid
+
+
+def define_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--project-uuid", type=is_published_uuid, required=True,
+                        help="UUID of the project to be patched. Needs to be published in the DCP.")
+    parser.add_argument("-d", "--dry-run", action="store_true", default=False, help="Dry-run flag. No PATCH operations"
+                                                                                    "will be performed.")
+
+    return parser
+
+
+def build_azul_query(project_uuid: str):
+    """
+    Build a ElasticSearch query to extract all metadata related to project with UUID "project_uuid"
+    :param project_uuid:
+    :return query:          string dump of query
+    """
+    query = {
+        "projectId": {
+            "is":
+                [
+                    project_uuid
+                ]
+        }
+    }
+    return json.dumps(query)
+
+
+def get_files_metadata_from_azul(query: str) -> list:
+    """
+    Retrieve file metadata from azul
+    :param query:           Query in str format to pass as a filter
+    :return files:          List of file metadata
+    """
+    # Build URL/endpoint and request metadata
+    azul_base = "https://service.azul.data.humancellatlas.org/"
+    azul_manifest_request_endpoint = "index/files"
+    r = rq.get(f"{azul_base}{azul_manifest_request_endpoint}?filters={query}&size=100")
+    r.raise_for_status()
+    r = r.json()
+
+    # Document ID (file metadata UUID) is outside of ["hits"]["files"], so some data massage
+    def change_datafile_uuid(azul_hit: dict):
+        azul_hit['files'][0]['uuid'] = azul_hit['entryId']
+        return azul_hit
+    files = list(map(change_datafile_uuid, r['hits']))
+
+    # A bit more data massage to return only the file metadata (Easier parsing for later)
+    files = [hit['files'][0] for hit in files]
+
+    return files
+
+
+def parse_azul_metadata(azul_file_list: list) -> dict:
+    """
+    Parse the azul metadata about the files and return a map {uuid: metadata}, excluding DCP1/2 analysis files
+    :param azul_file_list:          File list from azul, output from get_files_metadata_from_azul()
+    :return file_metadata_by_uuid:  Dictionary in the format {file_uuid:file_metadata} for all files in azul_file_list
+    """
+    uuid_metadata_map = {file['uuid']: file for file in azul_file_list}
+
+    filesource_invalid_values = ["HCA Release", "DCP/2 Analysis", "DCP/1 Matrix Service"]
```

Comment:
I would add this also at the beginning of the file (same comment as the azul hostname)

Reply:
?

Reply:
I don't understand what you mean by this comment.

You mean to convert this into a global variable? It's only used in this function

The only reason I made it into a variable and not into the dictionary below was for readability

Reply:
ok, but see my next comment